Sources/SwiftFusion/Geometry/Rot2.swift

```diff
@@ -135,12 +135,20 @@ public extension Rot2 {
     Point2(c * p.x + -s * p.y, s * p.x + c * p.y)
   }
 
+  func rotate(_ p: Point2.TangentVector) -> Point2.TangentVector {
```

Comment:
Hmm, SO(2) typically acts on Point2, so is this a *second* rotate for Point2.TangentVector? Of course, they are the same, because Point2 is a vector space. in fact, in GTSAM we had a special Point2 type at first but we deprecated it in favor of just equating it to Vector2. In which case we do not need a “special” rotate for TangentVectors. (see VectorSpace.h in GTSAM)

Reply:
I won't need these after I stop using `Point2.TangentVector` inside of `Pose2.TangentVector`, so I'll remove them!

Reply:
Still, I think my comments holds. I think we should simply identify Point2 with Vector2. Of course, that assumes we have a generic Vector class :-)

Reply:
Yes, I'll file a ticket for a vector type :)

Reply:
Can we just use 1D tensors?

Reply:
Do you mean plain unwrapped `Tensor` from `TensorFlow`?

That would work, but it has no static shape information, which would make the geometry apis more confusing/errorprone because it's harder to see what size vector you have and what size vector you need.

So I would prefer having something like Vector2, Vector3, ..., Vectorn, and Vector. (Unfortunately Swift doesn't have any way of being generic over integer values right now.) These could be wrappers around `Tensor`.

Reply:
Sure. FYI in GTSAM `Vector3` is a typedef to a fixed-size Eigen Vector. Having Vector2 and Vector3 here would be very familiar :-)